grl/environment/pomdp_file.py

```diff
@@ -390,9 +405,96 @@ def __get_pi_phi(self, i):
 
         return i + 1
 
+    def convert_obs_actions(self):
+        """
+        If we're here, we want to convert our POMDP phi-action tensor
+        into a phi tensor (so they're all the same).
+
+        To do so, we add an extra "initialization" observation
+        We also need to expand our state space to include the last action.
+
+        ONE BIG ASSUMPTION: In the POMDPs with phi(s, a), the action
+        is applied to the state BEFORE an observation comes out.
+        """
+        # first we construct our new transition function.
+        # we go from |A| x |S| x |S| -> |A| x (|S| + 1)|A| x (|S| + 1)|A|
+        # states are ordered as s0a0, s0a1, s0a2, ..., s1a0, s1a1, ..., etc.
+        n_actions = self.T.shape[0]
+        og_n_states = self.T.shape[1]
+
+        # T_extra_start is |A| x (|S| + 1) x (|S| + 1)
```

Comment:
Create a new (initial) state and concatenate it to T such that all the actions that "precede" the initial state lead to the same original start state distribution.